src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonRepository.kt

```diff
@@ -1,11 +1,27 @@
 package uk.gov.justice.digital.hmpps.prisonregister.model
 
+import org.springframework.data.jpa.repository.Query
 import org.springframework.data.repository.CrudRepository
+import org.springframework.data.repository.query.Param
 import org.springframework.stereotype.Repository
 
 @Repository
 interface PrisonRepository : CrudRepository<Prison, String> {
   fun findByActiveOrderByPrisonId(active: Boolean): List<Prison>
 
   fun findByGpPracticeGpPracticeCode(gpPracticeCode: String): Prison?
+
+  @Query(
+    """
+    select p from Prison p where p in (
+      select distinct p from Prison p
+      where (:active is null or p.active = :active)
+      and (:textSearchUpperCase is null or (UPPER(p.prisonId) like %:textSearchUpperCase% or UPPER(p.name) like %:textSearchUpperCase%))
```

Comment:
Had attempted to do `like UPPER(%:textSearch%)` but couldn't get it to work, so instead named the parameter  `textSearchUpperCase` and passed it as uppercase from `PrisonService`

Reply:
Would we be better having a Filter? as this could get quite unruly ?
Have a look at UserFilter in hmpps-auth

Reply:
Yes filter needed

Reply:
That's much neater! Added `PrisonFilter`